minio/minio
# Use the B2 'list' endpoint to determine file ID

## Description
Switches the B2 gateway to retrieve a list of files limited to 1 in order to find the ID of a file rather than attempting to download the first byte of the file.

**Note:** This is my first Go attempt so code-style suggestions more than welcome.

## Motivation and Context
This is related to issue https://github.com/minio/minio/issues/8166.

I determined that the reason for the error is that the B2 gateway was attempting to download the first byte of the file to determine the Backblaze file ID, with 0 byte files this will obviously not work so this PR switches it to use an alternative method.

## How to test this PR?

- Run MinIO with the B2 gateway installed
- Upload a 0 byte file (can be created with `touch` locally)
- Attempt to download the file from the MinIO UI
- Previously this would have failed, should now work

I've tested this locally too by uploading a 0 byte file to B2 through MinIO and downloading again successfully.

## Types of changes

## Checklist:
